tests/unit/checkboxradio/checkboxradio_core.js

```diff
@@ -0,0 +1,154 @@
+		label = $("label[for=check]");
+	ok( input.is( ":visible" ) );
+	ok( !label.hasClass(".ui-button)") );
+	input.checkboxradio();
+	equal( input.is( ".ui-helper-hidden-accessible.ui-checkboxradio" ), true,
+		"Input has proper classes" );
+	equal( label.is( ".ui-icon-beginning.ui-button.ui-widget.ui-checkboxradio-label.ui-corner-all" ),
+		true, "Label has proper classes" );
+});
+
+test("Radios", function() {
+	expect( 4 );
+	var inputs = $("#radio0 input"),
+		labels = $("#radio0 label");
+	ok( inputs.is(":visible") );
+	ok( labels.is(":not(.ui-button)") );
+	inputs.checkboxradio();
+	ok( inputs.is(".ui-helper-hidden-accessible") );
+	ok( labels.is(".ui-button") );
+});
+
+function assert(noForm, form1, form2) {
+	ok( $("#radio0 .ui-button" + noForm).is(".ui-state-active") );
+	ok( $("#radio1 .ui-button" + form1).is(".ui-state-active") );
+	ok( $("#radio2 .ui-button" + form2).is(".ui-state-active") );
+}
+
+test("radio groups", function() {
+	expect( 12 );
+	$("input[type=radio]").checkboxradio();
+	assert(":eq(0)", ":eq(1)", ":eq(2)");
+
+	// click outside of forms
+	$("#radio0 .ui-button:eq(1)").simulate("click");
+	assert(":eq(1)", ":eq(1)", ":eq(2)");
+
+	// click in first form
+	$("#radio1 .ui-button:eq(0)").simulate("click");
+	assert(":eq(1)", ":eq(0)", ":eq(2)");
+
+	// click in second form
+	$("#radio2 .ui-button:eq(0)").simulate("click");
+	assert(":eq(1)", ":eq(0)", ":eq(0)");
+});
+
+asyncTest( "Checkbox/Radiobutton do not Show Focused State when using Keyboard Navigation", function() {
+	expect( 2 );
+	var check = $( "#check" ).checkboxradio(),
+		label = $( "label[for='check']" );
+	ok( !label.is( ".ui-state-focus" ) );
+	check.focus();
+	setTimeout(function() {
+		ok( label.is( ".ui-state-focus" ) );
+		start();
+	});
+});
+
+// TODO: simulated click events don't behave like real click events in IE
+// remove this when simulate properly simulates this
+// see http://yuilibrary.com/projects/yui2/ticket/2528826 fore more info
+if ( !$.ui.ie || ( document.documentMode && document.documentMode > 8 ) ) {
+	asyncTest( "Ensure checked after single click on checkbox label button", function() {
+		expect( 2 );
+
+		$( "#check2" ).checkboxradio().change( function() {
+			var label = $( this ).checkboxradio( "widget" );
+			ok( this.checked, "checked ok" );
+
+			// The following test is commented out for now because with new markup we are trying to avoid aria
+			//ok( lbl.attr("aria-pressed") === "true", "aria ok" );
+			ok( label.hasClass( "ui-state-active" ), "ui-state-active ok" );
+		});
+
+		// Support: Opera
+		// Opera doesn't trigger a change event when this is done synchronously.
+		// This seems to be a side effect of another test, but until that can be
+		// tracked down, this delay will have to do.
+		setTimeout(function() {
+			$( "#check2" ).checkboxradio( "widget" ).simulate( "click" );
+			start();
+		});
+	});
+}
+test( "Checkbox creation that requires a matching finds label in all cases", function() {
```

Comment:
Not directly related, but it looks like there are also not enough tests for the various scenarios implemented by `_getRadioGroup`.